modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/pagemem/PagesWriteThrottlePolicy.java

```diff
@@ -39,11 +76,10 @@
     void onMarkDirty(boolean isPageInCheckpoint);
 
     /**
-     * Callback to try wakeup throttled threads.
+     * Callback to try wakeup throttled threads. Invoked when the Checkpoint Buffer use drops below a certain
+     * threshold.
      */
-    default void tryWakeupThrottledThreads() {
-        // No-op.
-    }
+    void tryWakeupThrottledThreads();
```

Comment:
I'm not the inventor of this name, but I think it is because it wakes the threads up conditionally:

```
if (!isCPBufferInDangerZone()) {
```

Reply:
but then those threads would not be throttled, no?

Reply:
Imagine the following situation:

1. Writes are very fast, so the CP Buffer is over 2/3
2. Writes get throttled, threads are parked
3. Meanwhile, the checkpointer does its job and offloads the CP Buffer so that it gets filled only 1/2
4. This is the moment when `tryWakeupThrottledThreads()` gets called. New writes are not throttled anymore, but there could still be some parked threads that were parked before (when the CP Buffer was still over 2/3).

So it seems reasonable to try and wake them up.

Reply:
yes, I understand that, I'm just confused that this method's name starts with `try` as if it may fail to wake up some threads

Reply:
How about just `wakeupThrottledThreads()`? Or, if we want to keep the flavor of conditionality, `wakeupThrottledThreadsIfCpBufferIsNotOverloaded()`?

Reply:
I'm all for `wakeupThrottledThreads`

Reply:
Renamed the method as suggested